lyluatex.lua

```diff
@@ -409,6 +409,36 @@ function Score:margins()
     end
 end
 
+function Score:optimize_pdf()
+    if self['optimize-pdf'] then
+        local pdf2ps, ps2pdf, pdf2pdf, path
+        for file in lfs.dir(self.tmpdir) do
```

Comment:
Maybe I'm not seeing the forest for the trees, but is it necessary to loop over all the files in the temp directory? Wouldn't it be sufficient to write something like "try to open `self.output..'.pdf`"?

Reply:
That depends on wether `fullpage` is set or not. I could do this as "try/fail" for each possible name (reading `-systems.count`), but I found it easier to do as is, though perhaps not as "cpu-optimized". I think that it shouldn't be noticed as soon as there aren't thousands of files in the temporary directory.